test/persistence/dbschema_test.cpp

```diff
@@ -54,192 +48,54 @@ private slots:
-    {"peers", "CREATE TABLE peers (id INTEGER PRIMARY KEY, public_key TEXT NOT NULL UNIQUE)"},
-    {"broken_messages", "CREATE TABLE broken_messages (id INTEGER PRIMARY KEY, reason INTEGER NOT "
-                        "NULL DEFAULT 0, FOREIGN KEY (id) REFERENCES history(id))"},
-    {"system_messages",
-     "CREATE TABLE system_messages (id INTEGER PRIMARY KEY, message_type CHAR(1) NOT NULL CHECK "
-     "(message_type = 'S'), system_message_type INTEGER NOT NULL, arg1 BLOB, arg2 BLOB, arg3 BLOB, arg4 BLOB, "
-     "FOREIGN KEY (id, message_type) REFERENCES history(id, message_type))"},
-    {"chat_id_idx", "CREATE INDEX chat_id_idx on history (chat_id)"}};
-
-
-void TestDbSchema::initTestCase()
+void TestDbSchema::init()
 {
-    for (const auto& path : testFileList) {
-        QVERIFY(!QFileInfo{path}.exists());
-    }
-    initSucess = true;
+    testDatabaseFile = std::unique_ptr<QTemporaryFile>(new QTemporaryFile());
+    // fileName is only defined once the file is opened. Since RawDatabase
+    // will be openening the file itself not using QFile, open and close it now.
+    QVERIFY(testDatabaseFile->open());
+    testDatabaseFile->close();
 }
 
-void TestDbSchema::cleanupTestCase() const
+void TestDbSchema::cleanup()
 {
-    if (!initSucess) {
-        qWarning() << "init failed, skipping cleanup to avoid loss of data";
-        return;
-    }
-    for (const auto& path : testFileList) {
-        QFile::remove(path);
-    }
-}
-
-void TestDbSchema::verifyDb(std::shared_ptr<RawDatabase> db, const std::vector<SqliteMasterEntry>& expectedSql)
-{
-    QVERIFY(db->execNow(RawDatabase::Query(QStringLiteral(
-        "SELECT name, sql FROM sqlite_master;"),
-        [&](const QVector<QVariant>& row) {
-            const QString tableName = row[0].toString();
-            if (row[1].isNull()) {
-                // implicit indexes are automatically created for primary key constraints and unique constraints
-                // so their existence is already covered by the table creation SQL
-                return;
-            }
-            QString tableSql = row[1].toString();
-            // table and column names can be quoted. UPDATE TEABLE automatically quotes the new names, but this
-            // has no functional impact on the schema. Strip quotes for comparison so that our created schema
-            // matches schema made from UPDATE TABLEs.
-            const QString unquotedTableSql = tableSql.remove("\"");
-            SqliteMasterEntry entry{tableName, unquotedTableSql};
-            QVERIFY(std::find(expectedSql.begin(), expectedSql.end(), entry) != expectedSql.end());
-        })));
-}
-
-void TestDbSchema::createSchemaAtVersion(std::shared_ptr<RawDatabase> db, const std::vector<SqliteMasterEntry>& schema)
-{
-    QVector<RawDatabase::Query> queries;
-    for (auto const& entry : schema) {
-        queries += entry.sql;
-    }
-    QVERIFY(db->execNow(queries));
+    testDatabaseFile.reset();
```

Comment:
Should this resize(0) as well? Docs just say that it seeks to the start, but I could imagine that reading a db with garbage on the end could still produce unexpected results.

Reply:
I'm not sure what you mean here? It's resetting the std::unique_ptr, destroying the QTemporaryFile, which deletes the temporary file altogether. Each test case is running with a brand-new QTemporaryFile since init() runs before each test.

Reply:
🤦 i misread as -> reset